pkg/watcher/queue_watcher.go

```diff
@@ -0,0 +1,84 @@
+package watcher
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/Gamebuildr/Dave/pkg/config"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/aws/aws-sdk-go/service/sqs/sqsiface"
+)
+
+// QueueWatcher is the aws sns implementation of the NotificationService
+type QueueWatcher struct {
+	client sqsiface.SQSAPI
+}
+
+var messageIDToReceiptHandle = map[string]*string{}
+
+// Setup gets the watcher ready to use
+func (watcher *QueueWatcher) Setup() {
```

Comment:
yea there's a weird error when adding it to an interface that makes it seem like it can't - but it can. I forgot the specific error and fix - I know it can be done - if just a bit weird.